test/cusparse.jl

```diff
@@ -711,17 +707,15 @@ end
             end
             @test_throws DimensionMismatch mm!('N','T',alpha,d_A,d_B,beta,d_C,'O')
             @test_throws DimensionMismatch mm!('T','N',alpha,d_A,d_B,beta,d_C,'O')
-            @test_throws DimensionMismatch mm!('T','T',alpha,d_A,d_B,beta,d_C,'O')
+            if !(CUSPARSE.version() < v"10.3.1" && d_A isa CuSparseMatrixCSR)
+                # A^T is not supported with B^T
+                @test_throws DimensionMismatch mm!('T','T',alpha,d_A,d_B,beta,d_C,'O')
```

Comment:
I wouldn't throw a DimensionMismatch here, but an `error` that indicates this is related to the CUDA version.